### PR TITLE
Python 3 compatibility

### PR DESCRIPTION
**Pull request by Neitsch**
Python 3 appears to be a lot fast, so I fixed all compatibility issues.
* Changed all print functions to be python 3 compatible
* Removed utf-8 encoding - it appears to work without

I'm sorry the diff is so big. Somehow jupyter added some meta data that I do not know how to remove.

### CODE REVIEW

File: sms_analysis.ipynb
Comment (mdezube) [inline sms_analysis.ipynb]: Why is this necessary in this commit?

File: sms_analysis.ipynb
Comment (mdezube) [inline sms_analysis.ipynb]: Why do we have to remove this?

**Review by mdezube — CHANGES_REQUESTED** [review]
Two high level comments:

1) Does the iphone_connector file not have to be edited?

2) You'll have to eliminate the noise around metadata{} from this commit...Just edit the .ipynb file directly in sublime or something, it's really just a glorified JSON file.

**Comment by Neitsch** [comment]
1) iphone_connector will have to be edited, but I don't have an iphone. Without that I'm not comfortable making a commit.
2) Ugh, okay. Will do once you say that you will approve afterwards. Jupyter will generate the metadata every time 😑

File: sms_analysis.ipynb
Comment (Neitsch) [reply sms_analysis.ipynb]: Wanted to make it similar to all other widgets, but you're right, this should be a separate PR

File: sms_analysis.ipynb
Comment (Neitsch) [reply sms_analysis.ipynb]: The problem appears to be that python 2 and 3 use string differently. You can not call `.encode('utf-8')` in python 3. I've tested removing it with python 2 and 3 without any problems.

File: sms_analysis.ipynb
Comment (mdezube) [inline sms_analysis.ipynb]: Generally this goes at the top, e.g.

```
from __future__ import print_function

import copy
import json
...
```

File: sms_analysis.ipynb
Comment (mdezube) [inline sms_analysis.ipynb]: To make this compatible add `from __future__ import division` to the top of the file and then do `//` instead of `/`, that truncates after division

File: sms_analysis.ipynb
Comment (mdezube) [inline sms_analysis.ipynb]: This is already a string, drop str()

File: sms_analysis.ipynb
Comment (mdezube) [reply sms_analysis.ipynb]: It throws an error for me in Python 2, so create a helper method at the top of the file that looks like:

```
# To aid with Python 2.7/3 compatibility,
convert_unicode_to_str_if_needed(unicode_or_str):
  if type(unicode_or_str).__name__ == 'unicode':
    return unicode_or_str.encode('utf-8')
  return unicode_or_str
```

And use this here and below.

**Review by mdezube — CHANGES_REQUESTED** [review]
Just a few comments here and there.  I also checked, and we can safely submit this and I can update the iphone_connector.py afterwards, we won't have any issues with them each being different, however python 3 won't be supported until I update it, which I'll do soon.

Go ahead and remove the other noise and I'll merge this after you address the comments I dropped.  If there's anything remaining I'll take care of it in a local client before merging.

File: sms_analysis.ipynb
Comment (Neitsch) [reply sms_analysis.ipynb]: Hm, that is weird, sorry about that! I guess the facebook_connector might be outputting a different string format than the iphone_connector. Might try to look into it a little more later.

**Comment by mdezube** [comment]
Also a tip:

Install [p4merge](https://www.perforce.com/product/components/perforce-visual-merge-and-diff-tools), set that as the difftool for github and then call it via `git difftool -y <file>`.  It's a much much better differ than the built in one, and will allow you to revert the noise with a quick click on each section instead of modifying it all by hand.

[SourceTree](https://www.sourcetreeapp.com/) is also great if you want a tool to better visualize github branches as well and it may support high quality diffs as well.

**Comment by mdezube** [comment]
Ahh I see this is ready for review, will get to it this week.  May just be a simple merge with a bit of cleanup.

**Pull request merged**
